### DISCUSSION TITLE
change 302 on 307

### DISCUSSION THREAD

**Discussion by redboxstudio** [keycloak/keycloak#28111] [Q&A]
Hello I get 405 when re-authorizing Ajax post requests, I send requests to auth keycloak 307 but the response from keycloak is 302, I tried replacing all Response.status(302).location with temporary redirect, but I still get 302, it turns out the redirect is not declared explicitly, like can I find it?
entry point here
https://github.com/keycloak/keycloak/blob/149e50e1b10aa7af975c761ba47b1fd8ab932d73/services/src/main/java/org/keycloak/protocol/oidc/endpoints/AuthorizationEndpoint.java#L105
somewhere here
https://github.com/keycloak/keycloak/blob/f970deac37d48972a9d8aa9cc94489202176a96b/services/src/main/java/org/keycloak/protocol/AuthorizationEndpointBase.java#L116

and I see this in the log
https://github.com/keycloak/keycloak/blob/f970deac37d48972a9d8aa9cc94489202176a96b/services/src/main/java/org/keycloak/authentication/AuthenticationProcessor.java#L1027
![image](https://github.com/keycloak/keycloak/assets/7893253/8480ae5c-e594-4867-b70f-cfa38dad8149)
and I see this in the log
https://github.com/keycloak/keycloak/blob/f970deac37d48972a9d8aa9cc94489202176a96b/services/src/main/java/org/keycloak/authentication/DefaultAuthenticationFlow.java#L227
https://github.com/keycloak/keycloak/blob/f970deac37d48972a9d8aa9cc94489202176a96b/services/src/main/java/org/keycloak/authentication/DefaultAuthenticationFlow.java#L441
https://github.com/keycloak/keycloak/blob/f970deac37d48972a9d8aa9cc94489202176a96b/services/src/main/java/org/keycloak/authentication/DefaultAuthenticationFlow.java#L501
https://github.com/keycloak/keycloak/blob/f970deac37d48972a9d8aa9cc94489202176a96b/services/src/main/java/org/keycloak/authentication/DefaultAuthenticationFlow.java#L540

**Comment by codespearhead** [comment]
Are you implementing an OIDC client by yourself?

If so, is that for learning purposes? Because there's a reason there's [few OIDC client implementations in the wild](https://openid.net/developers/certified-openid-connect-implementations/): they're hard to create and maintain.

[Keycloak has an official one](https://www.keycloak.org/docs/latest/securing_apps/index.html#_javascript_adapter) for client-side JavaScript.
